velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -489,11 +489,16 @@ std::unique_ptr<dwio::common::SerDeOptions> parseSerdeParameters(
     mapKeyDelim = parseDelimiter(mapKeyIt->second);
   }
 
-  uint8_t escapeChar;
-  bool hasEscapeChar = false;
-  if (escapeCharIt != serdeParameters.end() && !escapeCharIt->second.empty()) {
-    hasEscapeChar = true;
-    escapeChar = escapeCharIt->second[0];
+  // By default we have standard escape character as '\\'.
+  // Only when empty string is specified we don't have escape character.
+  uint8_t escapeChar = '\\';
+  bool hasEscapeChar = true;
+  if (escapeCharIt != serdeParameters.end()) {
+    if (!escapeCharIt->second.empty()) {
+      escapeChar = escapeCharIt->second[0];
+    } else {
+      hasEscapeChar = false;
+    }
   }
 
   auto serDeOptions = hasEscapeChar
```

Comment:
What's the issue here? If 'hasEscapeChar' is false, escapeChar is not used below?

Reply:
@xiaoxmeng 
Basically, Java reads a row properly, where one column has escaped \t characters, while Native treats the escaped characters as column/field separator and reads row wrong. This is when reading from a Text format file.

After some debugging I concluded that escape character is not in the list of serdeParameters and our code simply switches off the escaping.
This change is to fix that.

However, I'm still testing it in the shadow - some tweaks might be needed.

Reply:
@xiaoxmeng 

To correct: Java interprets empty escape character in serde params like a default `\`.
We still keep escaping off when escape character is not specified in serde params.
Velox current behavior is to switch escaping off when empty escape character is specified in serde params.

This change is needed to resolve correctness issue between Presto Native and Presto Java.

Reply:
escapeChar is not used in SerDeOptions if hasEscapeChar is false in current code. Shall we change that in this fix as well?

Reply:
@xiaoxmeng 
> escapeChar is not used in SerDeOptions if hasEscapeChar is false in current code. Shall we change that in this fix as well?

I this this by design.
Apache Hive has that logic too: the data might have no escaping whatsoever, so unescaping is not done in such case.
What we want to change in Velox is to understand SerDe params in the same way Apache Hive does.
This change is about that.

Reply:
Could be a follow up to change `SerDeOptions::escapeChar` to `optional<char>`

Reply:
We can use optional, however this is not a priority.